radix-engine/src/engine/process.rs

```diff
@@ -364,6 +364,10 @@ impl<'r, 'l, L: SubstateStore> Process<'r, 'l, L> {
             .remove(&proof_id)
             .ok_or(RuntimeError::ProofNotFound(proof_id))?;
 
+        if proof.is_restricted() {
```

Comment:
We can split the proofs into two groups: `transmissible_proofs` and `restricted_proofs`. It will make the move logic simpler but will make the read logic (Scrypto can reference both groups) harder. 

I have no strong opinion. I'm happy to go either way.

Reply:
Ha, let me try this out.